pyccel/ast/operators.py

```diff
@@ -568,12 +576,20 @@ def __new__(cls, arg1, arg2, simplify = False):
         if simplify:
             if isinstance(arg2, PyccelUnarySub):
                 return PyccelAdd(arg1, arg2.args[0], simplify = True)
-            elif isinstance(arg1, Literal) and isinstance(arg2, Literal):
+            else:
```

Comment:
Why did you remove this check?

Reply:
If I understand it right, this check insures that arg1 and arg2 are positive. I see there is no use to it as long as we are checking later if the result is positive or negative.

Reply:
I understand I should have it back (the tests don't work unless I have this check), I ll try to understand why

Reply:
> If I understand it right, this check insures that arg1 and arg2 are positive

No, this checks if arg1 and arg2 are literal values. That means it is a string, a (positive) float, a (positive) int, a complex or a boolean. But if you don't check then you also have to handle types such as `Variable`, `PyccelMod`, `NumpyRand`, etc. Basically anything that can be added. These other classes don't have a `python_value` property as we don't know their value and that value may not even be constant